Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py

```diff
@@ -0,0 +1,705 @@
+        index = indice_descending_severity[i]
+        if results[index].get('score') == SCORE_INVALID_URL:
+            continue
+        verdict = results[index].get('verdict')
+        pred_json = results[index].get('pred_json')
+        url = results[index].get('url')
+        is_white_listed = results[index].get('is_white_listed')
+        output_rasterize = results[index].get('output_rasterize')
+        summary_json = return_entry_summary(pred_json, url, is_white_listed, output_rasterize, verdict)
+        outputs.append(summary_json)
+        outputs = [x for x in outputs if x]
+    return outputs
+
+
+def save_model_in_demisto(model):
+    encoded_model = base64.b64encode(dill.dumps(model))  # guardrails-disable-line
+    res = demisto.executeCommand('createMLModel', {'modelData': encoded_model.decode('utf-8'),
+                                                   'modelName': URL_PHISHING_MODEL_NAME,
+                                                   'modelLabels': [MALICIOUS_VERDICT, BENIGN_VERDICT],
+                                                   'modelOverride': 'true',
+                                                   'modelHidden': True,
+                                                   'modelType': 'url_phishing',
+                                                   'modelExtraInfo': {
+                                                       OOB_MAJOR_VERSION_INFO_KEY: model.major,
+                                                       OOB_MINOR_VERSION_INFO_KEY: model.minor}})
+    if is_error(res):
+        return_error(get_error(res))
+
+
+def extract_urls(text):
+    res = demisto.executeCommand("extractIndicators", {"text": text})
+    if is_error(res):
+        return_error(get_error(res))
+    return list(set(json.loads(res[0]["Contents"]).get("URL", [])))
+
+
+def load_demisto_model():
+    model_64_str = get_model_data(URL_PHISHING_MODEL_NAME)[0]
+    model = decode_model_data(model_64_str)
+    return model
+
+def get_final_urls(urls, max_urls, model):
+    final_url = []
+    seen = []
+    low_priority_urls = []
+    i = 0
+    for url in urls:
+        if i < max_urls:
+            if extract_domainv2(url) in seen or extract_domainv2(url) in model.top_domains:
+                low_priority_urls.append(url)
+            else:
+                final_url.append(url)
+                seen.append(extract_domainv2(url))
+                i += 1
+    if len(final_url) < max_urls:
+        final_url = final_url + low_priority_urls[:min(len(low_priority_urls), max_urls-len(final_url))]
+    return final_url
+
+
+def extract_embedded_urls_from_html(html):
+    embedded_urls = []
+    soup = BeautifulSoup(html)
+    for a in soup.findAll('a'):
+        if a.has_attr('href'):
+            if a['href'] not in a.get_text():
+                embedded_urls.append(a['href'])
+    return embedded_urls
+
+
+def main():
+    msg_list = []
+    exist, demisto_major_version, demisto_minor_version = oob_model_exists_and_updated()
```

Comment:
when you check if the model exists - please also get the model data itself, so you won't need to call "getMLModel" again

Reply:
done